src/main/java/org/broadinstitute/hellbender/tools/longreads/graph/AlignedBaseGraphCollection.java

```diff
@@ -0,0 +1,461 @@
+                        vertex = graphVertex;
+                        vertexIsInGraph = true;
+
+                        // If we added our last node, we need to make a link from the newly-added last node to
+                        // the current graph node.
+                        if ( lastNodeWasAddedToGraph ) {
+                            // This if statement can probably be removed:
+                            if (lastVertex.equals(vertex)) {
+                                logger.error("Equal Vertices detected: " + vertex.toString() + " @ " + vertex.getPos().toString());
+                                throw new GATKException("STILL BUSTED, BUSTER!");
+                            }
+                            graph.addEdge(lastVertex, vertex);
+                        }
+                        lastNodeWasAddedToGraph = false;
+                        break;
+                    }
+                }
+            }
+
+            // We must add in the new vertex and its edges:
+            if ( !vertexIsInGraph ) {
+                // Add our vertex to the graph:
+                graph.addVertex(vertex);
+
+                // Add the vertex to our positional data store:
+                if ( positionVertexMap.containsKey(vertex.getPos()) ) {
+                    positionVertexMap.get(vertex.getPos()).add(vertex);
+                }
+                else {
+                    positionVertexMap.put(
+                            vertex.getPos(),
+                            new HashSet<>(Collections.singletonList(vertex))
+                    );
+                }
+
+                // Add an edge from the previous position if we must:
+                if (lastVertex != null) {
+                    graph.addEdge(lastVertex, vertex);
+                }
+                lastNodeWasAddedToGraph = true;
+            }
+
+            lastVertex = vertex;
+        }
+    }
+
+    /**
+     * Create a linear sequence of node objects based on aligned sequence information.
+     * @param bases Bases to be added in the order in which they appear in the read.
+     * @param cigar {@link List<CigarElement>} containing alignment information for each base in {@code bases}.
+     * @param contig The contig to which the given base sequence is aligned.
+     * @param startPos The start position of the first aligned base in the given base sequence.
+     * @param readName The name of the parent read from which this aligned sequence information is derived.
+     * @return A {@link Queue<AlignedBaseVertex>} representing a linear graph of bases in a given read.
+     */
+    private Queue<AlignedBaseVertex> createAlignedNodes(final byte[] bases, final List<CigarElement> cigar, final String contig, final int startPos, final String readName) {
+
+        final Queue<AlignedBaseVertex> nodeQueue = new ArrayDeque<>();
+
+        int readBasePos = 0;
+        int insertionOffset = 0;
```

Comment:
Can this variable's scope be inside the `for` loop -- and then you could initialize to zero as the first line in the loop without needing to reset to zero at the end of the loop?

Reply:
It looks that way.  Fixed.